mold.h

```diff
@@ -40,6 +40,16 @@
 # define unreachable() assert(0 && "unreachable")
 #endif
 
+#if defined(__has_feature)
+# if __has_feature(thread_sanitizer)
+#  define MOLD_TSAN_DISABLE __attribute__((no_sanitize("thread")))
```

Comment:
Does unrecognized `__attribute__` cause an error for gcc/clang? If not, we want to directly use this without macro.